pocs/sensors/sensor_recorder.py

```diff
@@ -0,0 +1,111 @@
+import queue as queue_module
+import threading
+
+
+class SensorRecorder(threading.Thread):
+    """Monitors a queue for readings from sensors, records in the store.
+
+    This is a Thread sub-class. This implies that it can only be started
+    running once (threading.Thread does not allow restarting).
+
+    This is the format in which we expect entries in the queue:
+        {'name': board_name, 'timestamp': timestamp_string, 'data': reading}
+    """
+
+    def __init__(self,
+                 save_func,
+                 send_func,
+                 logger,
+                 max_queue_size=100,
+                 queue_read_timeout=2.0,
+                 **kwargs):
+        """
+        Args:
+            save_func:
+                (Optional) Function to save a single sensor reading to a data store.
+                Takes two parameters, name and reading.
+            send_func:
+                (Optional) Function to send each reading to the messaging system.
+                Takes two parameters, name and reading.
+            logger:
+                Logger to use.
+            max_queue_size:
+                Number of readings to allow to accumulate in the queue.
+            queue_read_timeout:
+                Number of seconds to wait for a reading before checking
+                whether stop() has been called.
+            **kwargs: Any other parameters to threading.Thread.
+        """
+        kwargs.setdefault('name', 'SensorRecorder')
+        kwargs.setdefault('daemon', True)
+        assert 'target' not in kwargs
+        kwargs['target'] = None
+        super().__init__(**kwargs)
+
+        # These are all private and should only be accessed by the public methods.
+        self._save_func = save_func
+        self._send_func = send_func
+        self._logger = logger
```

Comment:
I thought I made a comment on this before, but don't think this logger should be private for same reasons as mentioned about logger above (and especially if you will make a stdout logger)

Reply:
I don't understand this comment about _logger being private. Can you elaborate?

Reply:
I was just wondering why it is private as our `logger` is not private anywhere else.

Reply:
In the case of logger, there isn't a strong reason except that it must not be replaced from outside while it is in use by the Thread. So, I wanted to make sure that all the fields are treated as private, rather than have mutexes for some public fields and not for other public fields (i.e. to avoid confusion).

I suppose this also reflects my decades of C++ & Java development (30 years for C++), where we  default to making all fields private unless there is an amazingly good reason to make them public. In this case, what we want from this class is its behavior rather than access to its fields... except the queue.

Reply:
Ok, not a problem. I was mostly pushing for consistency reasons but since this isn't used directly from within a POCS instance that's not as important.